### ISSUE TITLE
Automatically assigned groups are not used for submission permissions

### ISSUE THREAD

**Issue by olli-gold** [DSpace/DSpace#8939]
**Describe the bug**
In DSpace it is possible to assign submission permissions to a collection to users or to subgroups. It seems, that the subgroups are not used at all to permit a user to submit anything, unless the user is assigned to this subgroup directly. So it is impossible to allow a group of users for submission in a collection without adding every single user to do the submission.

**To Reproduce**
1. Create a submission permission for a collection
2. Create a group g
3. Set the group g as a subgroup to the submission permission
4. Setup a configuration, where all users, who log in, will be assigned to the newly created group g (`authentication-password.login.specialgroup = g` in `local.cfg`)
5. Log in as a non-administrative user
6. Check in the user profile, if the user is set as a member of group g
7. Try to start a submission in the collection

**Expected behavior**
The user should be able to submit a new item to the collection.

**Related work**
I'm not aware of any related work.

**Comment by tdonohue** [comment]
@olli-gold :  I just tried to reproduce this in DSpace 7.6 and I **cannot reproduce it**.

Here's what I did:

1. Created a new Group named `LoggedInUsers`.  Verified that group is completely empty.
2. Created a Collection. Edited it, from the "Assign Roles" tab, I created a new "Submitters" group.  This created a group named `COLLECTION_[uuid]_SUBMIT`.  To that group, I added the subgroup `LoggedInUsers`.  No other users or groups were added.
3. On backend now, I added this to my `local.cfg`: `authentication-password.login.specialgroup = LoggedInUsers`
4. Now, logged out and logged in as a user with no special privileges (no group memberships)
5. From the Account Profile page for that user, I see it lists `LoggedInUsers` in the "Authorization special groups you belong to"
6. The sidebar is visible, and I can see the "New -> Item" option.  Click it.  Select the Collection created in step two.
7. Fill out submission form & submit.  Everything works.

I believe this bug either was already fixed, or perhaps you've misconfigured something on your side?

Moving this back to the "Backlog board" and flagging as not reproducible for now.
